### PR TITLE
[h2o] update h2o version and deprecate Dataproc 1.5

### PR DESCRIPTION
**Pull request by bradmiro**
Removed support for 1.5 due to #774 . Now fails with an error message asking the user to use 1.3, 1.4 or 2.0+

Updated README to reflect this. 

Updated h2o version.

Fixed test script.

### CODE REVIEW

File: h2o/h2o.sh [hunk 1]
```diff
@@ -54,6 +54,11 @@ EOF
 }
 
 function main() {
+  if [[ "${DATAPROC_VERSION}" == "1.5" ]]; then
```
Comment (medb) [inline h2o/h2o.sh:57]: Usually we put such conditions at very top of the init action: https://github.com/GoogleCloudDataproc/initialization-actions/blob/85e6216b9c0d9c6ea2f14b728b47862ab0e1bec1/conda/install-conda-env.sh#L4

File: h2o/h2o.sh [hunk 2]
```diff
@@ -54,6 +54,11 @@ EOF
...
+    echo "Dataproc 1.5 not supported. Please use Dataproc 1.3, 1.4 or 2.0+"
```
Comment (medb) [inline h2o/h2o.sh:58]: I think we need to add similar condition to skip test for 1.5 images, if we have it?

**Review by medb — CHANGES_REQUESTED** [review]

File: h2o/h2o.sh (hunk 1 above)
Comment (bradmiro) [reply h2o/h2o.sh]: Fixed

File: h2o/h2o.sh (hunk 2 above)
Comment (bradmiro) [reply h2o/h2o.sh]: Fixed

**Comment by medb** [comment]
Seems like you need to rebase (not merge) on top of master to fix presubmit.

**Comment by bradmiro** [comment]
Think that should have fixed it.

**Comment by bradmiro** [comment]
Skip logic was flawed; fixed.

**Comment by medb** [comment]
Seems like tests are failing for older images

**Comment by bradmiro** [comment]
I made a change to the test script that may have fixed the issue.

**Comment by medb** [comment]
Plase, revert RAPIDS init action changes.

**Comment by bradmiro** [comment]
Sorry about that, just rebased and removed the commit.

File: h2o/sample-script.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 from pysparkling import *
 
 spark = SparkSession.builder.appName("SparklingWaterApp").getOrCreate()
-hc = H2OContext.getOrCreate(spark)
+hc = H2OContext.getOrCreate()
```
Comment (medb) [inline h2o/sample-script.py:6]: Should we revert this?

**Review by medb — CHANGES_REQUESTED** [review]

File: h2o/sample-script.py (hunk 3 above)
Comment (bradmiro) [reply h2o/sample-script.py:6]: I didn't think this was necessary but I'll add it back and see if that helps.

**Comment by bradmiro** [comment]
These tests have been flaky for me locally with the same error that showed up here. I added the spark context back, can we try running again and see what happens?

File: h2o/sample-script.py (hunk 3 above)
Comment (medb) [reply h2o/sample-script.py]: It still failing, maybe try to increase cluster nodes size? By default they are configured to `e2-standard-2` machine type.

Comment (bradmiro) [reply h2o/sample-script.py]: Done.

Comment (medb) [reply h2o/sample-script.py]: Still does not work, maybe let's ignore all image versions except 2.0?

Comment (bradmiro) [reply h2o/sample-script.py]: Work for me, I'll make the change

Comment (bradmiro) [reply h2o/sample-script.py]: Just pushed

File: h2o/README.md [hunk 4]
```diff
@@ -15,34 +17,17 @@ initialization actions in production.
...
 
     ```bash
     REGION=<region>
     CLUSTER_NAME=<cluster_name>
     gcloud dataproc clusters create ${CLUSTER_NAME} \
-        --image-version 1.3 \
-        --metadata 'H2O_SPARKLING_WATER_VERSION=3.28.0.1-1' \
+        --image-version preview-ubuntu \
         --scopes "cloud-platform" \
         --initialization-actions "gs://goog-dataproc-initialization-actions-${REGION}/conda/bootstrap-conda.sh,gs://goog-dataproc-initialization-actions-${REGION}/h2o/h2o.sh"
```
Comment (medb) [inline h2o/README.md:28]: Remove conda init action.

**Review by medb — CHANGES_REQUESTED** [review]

File: h2o/sample-script.py (hunk 3 above)
Comment (medb) [reply h2o/sample-script.py]: Seems like it does not work even on 2.0

Comment (bradmiro) [reply h2o/sample-script.py]: Seemed like it was a "dataproc internal error", maybe a flake? This has consistently passed for me locally on 2.0.

File: h2o/README.md (hunk 4 above)
Comment (bradmiro) [reply h2o/README.md]: Done

File: h2o/sample-script.py (hunk 3 above)
Comment (medb) [reply h2o/sample-script.py:6]: Seems like presubmit passed for all image version except 1.5 after https://github.com/GoogleCloudDataproc/initialization-actions/pull/811

**Review by medb — APPROVED** [review]

**Pull request merged**